conf/varnish.vcl-example

```diff
@@ -0,0 +1,47 @@
+# A really basic example to use the Varnish api key checking.
+# Copy this to /etc/varnish/default.vcl to use it (or edit
+# /etc/default/varnishto point at this)
+import std;
+include "/etc/varnish/varnish-apikey.vcl";
+
+backend local_django {
+  .host = "127.0.0.1";
+  .port = "8000";
+}
+
+#
+# Subroutine that defines the api, key and token.
+#
+sub recognize_apiname_apikey_token {
+  # Identify api
+
+  # As mapit is both an API and a website, we only want to invoke the api key
+  # checking on the actual api endpoints. We purposely exclude the .html
+  # versions of the endpoints too, because they're basically documentation,
+  # not an actual use of the API.
+  if (req.url ~ "^/postcode/.*[^\.html$]" ||
+    req.url ~ "^/point/.*[^\.html$]" ||
+    req.url ~ "^/area/.*[^\.html$]" ||
+    req.url ~ "^/areas/.*[^\.html$]" ||
+    req.url ~ "^/nearest/.*[^\.html$]" ||
+    req.url ~ "^/code/.*[^\.html$]" ||
+    req.url ~ "^/generations/.*[^\.html$]") {
+    set req.http.apiname = "mapit";
+
+    # Save apikey if there is one
+    if (regsub(req.url, "^.*(\?|&)apikey=([^&;]*).*", "\2") != req.url) {
```

Comment:
Better as `[?&]apikey=([^&;]*)` – no need for "match anything" at start and end.